### ISSUE TITLE
🐛 BUG: Daily Streak Not Updating After Reading Post

### ISSUE THREAD

**Issue by Silverwolfxp** [dailydotdev/daily#1619]
### What went wrong? 🤔

I've been on daily.dev for almost a month now and haven't had any issues other than my streak not updating when I read a post like it's supposed to, and it says I've lost my streak. 
![Screenshot_20241126_013340](https://github.com/user-attachments/assets/5b38c8ba-50b7-4a82-9e5d-9899997b1f64)

### Expected Behavior

I expect it to update when I read a post

### Steps to Reproduce Issue

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See the error
```

### Browsers

Firefox

### OS

Linux

### Code of Conduct

- [X] I follow the conditions of this project.

**Comment by idoshamun** [comment]
Hi @Silverwolfxp,

Can you please make sure your profile timezone matches your real timezone?
https://app.daily.dev/account/notifications

**Comment by LaraGastaldi** [comment]
+1 here

**Comment by idoshamun** [comment]
We're looking into the issue

**Comment by Avendor7** [comment]
I just want to chime in and say that my streaks are calculated properly but the display on the site is off by one day. My reading for today is showing up for yesterday. 

![image](https://github.com/user-attachments/assets/1454df79-b76f-4a66-989e-9468a2d11d81)

**Comment by rebelchris** [comment]
Update:
We found a lead where we on the FE don't match the timezone correctly in some edge-cases.
Hoping to ship a fix soon. 🙏

**Comment by Kennethch-02** [comment]
Hi team, I lost my streak days due to this timezone issue. Is it possible to restore them? I was really motivated tracking my daily learning progress and maintaining the streak. Let me know if you need any additional information from my side to help resolve this.

**Comment by bobbyiliev** [comment]
I am also hitting this bug, traveling this week so I am in a different timezone.

**Comment by idoshamun** [comment]
@bobbyiliev have you updated your timezone accordingly?

**Comment by bobbyiliev** [comment]
> @bobbyiliev have you updated your timezone accordingly?

Yep, changed it already but it still does not seem to be working 
<img width="626" alt="image" src="https://github.com/user-attachments/assets/68a09b26-c646-4447-9959-d089c696b773">


It seems like it calculates the read articles for the previous day instead of the current one 
<img width="724" alt="image" src="https://github.com/user-attachments/assets/8beeffa0-6f5c-44fa-820a-968c77328010">

**Comment by idoshamun** [comment]
Thanks for letting us know. We're investigating the issue, may take us some time since it's holiday season now 😓 
We'll do our best to solve in a timely manner

**Comment by bobbyiliev** [comment]
In case this helps, I am back in Europe and this seems to have started working again as expected even though I had GTM -5 as my Daily.Dev timezone.

**Comment by capJavert** [comment]
Hello everyone, as you all know timezones and time are the one of the hardest things in programming but we worked hard to fix it. So we implemented following mechanisms to ensure your streak displays and counts correctly each day:
- we always show your streak days and reading status in your timezone that is [selected in your settings](https://app.daily.dev/account/notifications)
- since timezone in settings is detected during signup we also added a label to show which timezone you currently have (1. picture below)
- to prevent your streak showing wrong while traveling or if your timezone setting is invalid we added a new warning which will show in those cases (2. picture below)
  - when warning shows you have an option to go to settings and update your timezone to match your device or ignore it for that timezone (for example in case of temporary travel)

All in all, we hope this changes will help you keep your streak and reading latest news from development world consistent.

FYI: one note, currently this is only available in the webapp (https://app.daily.dev/) since our new extension version is still pending release on Chrome web store. I will notify here when it is released as well.

Picture 1. - new timezone label
<img width="480" alt="Image" src="https://github.com/user-attachments/assets/febd33d3-afdf-4681-8cb8-ce9c84f50076" />

Picture 2. - warning for timezone mismatch, clicking on label gives you option to update/ignore
<img width="476" alt="Image" src="https://github.com/user-attachments/assets/07eef3f6-57f6-4c86-947d-0df589685ea6" />

Let us know if you encounter any issues.
